src/main/java/com/codecool/virtualpub/ui/Input.java

```diff
@@ -31,13 +31,14 @@ private String getInput(){
     private int getChoice(Bartender bartender, String[] actions){
         while (true) {
             try {
+                display.displayMessage("\nEnter choice: ");
                 int input = Integer.parseInt(getInput());
-                if (input > actions.length) {
+                if (input > actions.length || input < 1) {
                     throw new Exception();
```

Comment:
Throwing an exception is not a bet idea, but then you should specify which kind of exception you want to throw. There are a lot of built-in exceptions in Java, try finding one that most describes the problem. But if the input is not valid we should just ask for a new one, we don't have to throw an exception. Try to avoid infinite loops like while(true) in your ode.